lib/matplotlib/tests/test_figure.py

```diff
@@ -1243,11 +1244,100 @@ def test_share_all(self):
                           ["."]]]]]
         ]
         fig = plt.figure()
-        ax_dict = fig.subplot_mosaic(layout, sharex=True, sharey=True)
+        ax_dict = fig.subplot_mosaic(layout, sharex=sharex, sharey=sharey)
         ax_dict["A"].set(xscale="log", yscale="logit")
         assert all(ax.get_xscale() == "log" and ax.get_yscale() == "logit"
                    for ax in ax_dict.values())
 
+    @check_figures_equal(extensions=["png"])
+    def test_sharex_row(self, fig_test, fig_ref):
```

Comment:
consider doing the test similar to the `test_share_all` function above, which sets the scale of shared axes and asserts that all expected sharing is done.

Specifically, without anything actually plotted, there is no way for `check_figures_equal` (which does an image comparison) to know whether two things are shared or not, as each will default to a range of 0-1 anyway...

Reply:
Okay, I will add that. Because sharing axes affect tick labels, I'd like to keep the image comparison.